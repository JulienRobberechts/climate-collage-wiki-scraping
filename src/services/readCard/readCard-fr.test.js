/**
 * @jest-environment node
 */
const { readCards, getCardData } = require("./readCard");
const { getCardsFrReference } = require("../data-access/cardsRepo");

const getOneCard = async (cardNumber, lang = "fr") => {
  const cards = await getCardsFrReference(lang);
  return cards.find(({ cardNum }) => cardNum === cardNumber);
};

describe("readCard", () => {
  it("read carte_2", async () => {
    const cardNumber = 2;
    const lang = "fr";
    const card = await getOneCard(cardNumber, lang);
    const {
      wikiId,
      cardNum,
      wikiUrl,
      wikiInternalName,
      title,
      backDescription,
    } = await getCardData(card, lang);
    expect({
      wikiId,
      cardNum,
      wikiUrl,
      wikiInternalName,
      title,
      backDescription,
    }).toStrictEqual({
      wikiId: 12,
      cardNum: 2,
      wikiUrl: "/wiki/index.php?title=Fr-fr_adulte_carte_2_industrie",
      wikiInternalName: "Fr-fr adulte carte 2 industrie",
      title: "Industrie",
      backDescription:
        "L'industrie utilise des énergies fossiles et de l'électricité. Elle représente 40% des Gaz à Effet de Serre (GES).",
    });
  });
  it("read carte_9", async () => {
    const cardNumber = 9;
    const lang = "fr";
    const card = await getOneCard(cardNumber, lang);
    const pageData = await getCardData(card, lang);
    expect(pageData).toMatchSnapshot();
  });
  it("read carte_13", async () => {
    const cardNumber = 13;
    const lang = "fr";
    const card = await getOneCard(cardNumber, lang);
    const pageData = await getCardData(card, lang);
    expect(pageData).toMatchSnapshot();
  });
  it("read all cards 1-2", async () => {
    const lang = "fr";
    const allCards = await getCardsFrReference(lang);
    const cards = await readCards(allCards, 1, 2, lang);
    expect(cards.length).toBe(2);
    expect(cards).toMatchSnapshot();
  }, 99000);
});

describe.skip("readCard by batch", () => {
  it.skip("read all cards 1-10", async () => {
    const lang = "fr";
    const allCards = await getCardsFrReference(lang);
    const cards = await readCards(allCards, 1, 10, lang);
    expect(cards.length).toBe(10);
  }, 99000);
});
