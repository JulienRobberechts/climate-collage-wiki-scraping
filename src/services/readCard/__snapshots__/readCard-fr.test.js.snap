// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readCard read all cards 1-2 1`] = `
Array [
  Object {
    "backDescription": "C'est là que tout commence...",
    "cardNum": 1,
    "cardSet": 1,
    "explanation": "C'est par là que tout commence... C'est aussi par là que commence le 5ème rapport du GIEC, dans l'avant-propos de son résumé, il est mis en évidence que la science montre à présent avec 95 % de certitude que depuis le milieu du XXe siècle, l’activité humaine est la cause principale du réchauffement observé. Cette carte peut être considérée...soit comme la cause de l'ensemble des cartes secteurs économiques (Industrie, Utilisation des bâtiments, Transport, Agriculture), soit comme un titre de l'ensemble de ces cartes (qui sont alors regroupées dans une grosse “patate“).",
    "title": "Activités humaines",
    "wikiId": 4,
    "wikiInternalName": "Fr-fr adulte carte 1 activités humaines",
    "wikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_1_activit%C3%A9s_humaines",
  },
  Object {
    "backDescription": "L'industrie utilise des énergies fossiles et de l'électricité. Elle représente 40% des Gaz à Effet de Serre (GES).",
    "cardNum": 2,
    "cardSet": 2,
    "explanation": "Il s'agit de la fabrication de l'ensemble des biens de consommation. L'industrie regroupe un très grand nombre de secteurs industriels différents dont les plus importants en émissions de GES sont la papeterie, le ciment, l'acier, l'aluminium, la chimie. Pour réduire les émissions de l'industrie, la solution consiste à allonger la durée de vie des produits, voire à réduire leur consommation.",
    "title": "Industrie",
    "wikiId": 12,
    "wikiInternalName": "Fr-fr adulte carte 2 industrie",
    "wikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_2_industrie",
  },
]
`;

exports[`readCard read carte_9 1`] = `
Object {
  "backDescription": "Le CO2 n'est pas le seul Gaz à Effet de Serre. Il y a aussi le méthane (CH4) et le protoxyde d'azote (N2O) (qui viennent en très grande partie de l'agriculture), ainsi que quelques autres.",
  "cardNum": 9,
  "cardSet": 2,
  "explanation": "Les autres GES décrits ici sont le méthane et le protoxyde d'azote. En réalité, il y en quelques autres comme les HFC (fluides réfrigérants). Il y a dégagement de méthane dès qu'il y a une décomposition anaérobie (c’est-à-dire sans oxygène) :dans la panse des vaches qu'on appelle aussi le rumen et qui donne son nom aux ruminants (dans le rumen, les bactéries \\"digèrent\\" la cellulose que la vache ne sait pas métaboliser, puis la vache régurgite cette herbe pour la mâcher et l'avaler pour de bon). dans les rizières car elles sont couvertes d'eau, et la matière organique immergée ne reçoit pas d'oxygène quand elle se décompose. dans les décharges de déchets, quand les tas sont trop grands pour que l'oxygène atteigne le fond du tas. Le méthane est aussi le composant principal du gaz naturel. Les fuites sur les gazoducs font donc aussi du méthane. Les principales sources d'émissions de méthane par les activités humaines (au niveau mondial) sont (chiffres 2003-2012) :Énergies Fossiles (production et utilisation) : ~34% des émissions anthropiques de CH4 dont : ~1/3 venant du charbon 2/3 venant du gaz naturel et du pétrole Ruminants et fumiers / lisiers : ~31% des émissions anthropiques de CH4 dont : 87% pour la fermentation entérique (éructation) ==> soit ~27% des émissions anthropiques de CH4 rien que pour ce poste 13% pour les fumiers et lisiers Décharges et gestion des déchets : ~18% des émissions anthropiques de CH4 Rizières : ~9% des émissions anthropiques de CH4 Combustion de biomasse : ~5% des émissions anthropiques de CH4 Combustion de biocarburant : ~3% des émissions anthropiques de CH4Le protoxyde d’azote (N2O) est principalement dû à l’utilisation d’engrais azotés agricoles, à la production d’aliments pour bétail et à certains procédés chimiques, comme la production d’acide nitrique. Il y a encore les gaz fluorés qui sont utilisés comme réfrigérants (climatisation et chaînes du froid), extincteurs et dans certains procédés industriels et biens de consommation (comme certains dissolvants). Ils ne sont pas naturellement présents dans l’atmosphère.",
  "title": "Autres GES",
  "wikiId": 26,
  "wikiInternalName": "Fr-fr adulte carte 9 autres ges",
  "wikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_9_autres_ges",
}
`;

exports[`readCard read carte_13 1`] = `
Object {
  "backDescription": "L'effet de serre est naturel. D’ailleurs, le premier GES naturel est la vapeur d’eau. Sans l'effet de serre, la planète serait 33°C plus froide. Mais le CO2 et les autres GES dus à l'Homme augmentent cet effet de serre naturel, ce qui réchauffe le climat.",
  "cardNum": 13,
  "cardSet": 1,
  "explanation": "Sur cette carte, on voit des flèches de deux couleurs : Le orange pour l'énergie qui vient du soleil (UV, lumière visible et infrarouge de haute fréquence) et celle qui est réfléchie par effet albedo à la même fréquence. L'albedo est la capacité d'un corps à renvoyer la lumière (un corps noir a un albedo de 0, un miroir a un albedo de 1. La terre a un albedo moyen de 0,31). Le rouge pour les infrarouges de basses fréquences, émis par la terre qui est moins chaude que le soleil, ou retenus par l'effet de serre. L'effet de serre repose sur le fait que ce ne soit pas le même rayonnement entrant que le rayonnement sortant. Sur la droite, \\"-18°C\\", c'est la température qu'il ferait sur terre sans l'effet de serre et \\"15°C\\", c'est la température réelle aujourd'hui. Il faisait \\"14°C\\" en 1850, c'est à dire avant que l'activité humaine ne produise cet effet de serre additionnel.",
  "title": "Effet de serre additionnel",
  "wikiId": 34,
  "wikiInternalName": "Fr-fr adulte carte 13 effet de serre additionnel",
  "wikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_13_effet_de_serre_additionnel",
}
`;
